sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,94 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=5, normalize=True):
+    """Top k Accuracy classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top `k` labels predicted (ranked by predicted scores). Note that
+    multilabel classification case isn't covered here.
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels.
+
+    y_score : array-like of shape (n_samples, n_classes)
+        Target scores. These can be either probability estimates or
+        non-thresholded decision values as returned by :term:`predict_proba` or
+        :term:`decision_function`.
+
+    k : int, default=5
+        Number of guesses allowed to find the correct label.
+
+    normalize : bool, default=True
+        If `True`, return the fraction of correctly classified samples.
+        Otherwise, return the number of correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        The best performance is 1 with `normalize == True` and the number of
+        samples with `normalize == False`.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    In cases where two or more labels are assigned equal predicted scores
+    labels with smallest indices will be chosen first. Then the result might
+    be incorrect if the good label falls at the treshold because of that.
```

Comment:
```suggestion
    In cases where two or more labels are assigned equal predicted scores,
    the labels with the smallest indices will be chosen first. Then, the result might
    be incorrect if the correct label falls after the threshold because of that.
```